src/cargo/ops/fix.rs

```diff
@@ -87,11 +88,26 @@ pub struct FixOptions {
     pub broken_code: bool,
 }
 
-pub fn fix(ws: &Workspace<'_>, opts: &mut FixOptions) -> CargoResult<()> {
-    check_version_control(ws.gctx(), opts)?;
+pub fn fix(
+    gctx: &GlobalContext,
+    original_ws: &Workspace<'_>,
+    root_manifest: &Path,
+    opts: &mut FixOptions,
+) -> CargoResult<()> {
+    check_version_control(gctx, opts)?;
+
     if opts.edition {
-        check_resolver_change(ws, opts)?;
+        let specs = opts.compile_opts.spec.to_package_id_specs(&original_ws)?;
+        let members: Vec<&Package> = original_ws
```

Comment:
We do filter by the `--package` flag which is us just doing what the user told us.  I'm not seeing how presence of `[proiject]` causes a problem.  We also have a test showing that it works.

Reply:
Sorry, I don't seem to have been clear.

What I meant was that if the members in the workspace use `[project]` instead of `[package]`, then executing the command `cargo fix --edition` in the workspace does not modify the members.

I'm wondering if the changes should be done on all members at once.

Reply:
I think I'm still missing the problem case.

Could you create a test case to demonstrate it?

Reply:
If there is a workspace `ws` containing the member `hello`.

which:
```
# [ws]/Cargo.toml
[workspace]
members = ["hello"]

[project]     # Pay attention here 
name = "ws"
version = "0.1.0"
edition = "2021"
```

```
# [ws]/hello/Cargo.toml
[project]     # Pay attention here 
name = "hello"
version = "0.1.0"
edition = "2021"
```

After running `cargo fix --edition --allow-no-vcs` in the workspace. The `cargo.toml` of `ws` completes the modification, but the `cargo.toml` of member `hello` does not.

I think both should be modified to `[package]` here.

Reply:
I've written that up as a test case which is passing.

Yes, we aren't migrating `hello`'s `Cargo.toml` but we also aren't migrating `hello`s `src/lib.rs`.  Manifest and rust migration should be done hand-in-hand.  The user has to opt-in to migrating `hello` with the `--workspace` flag.  `cargo fix --edition --workspace` will migrate both packages, manifest and rust.

<details>

```rust
#[cargo_test]
fn migrate_project_to_package_ws() {
    let p = project()
        .file(
            "Cargo.toml",
            r#"
cargo-features = ["edition2024"]

[workspace]
members = ["hello"]

# Before project
[ project ] # After project header
# After project header line
name = "foo"
edition = "2021"
# After project table
"#,
        )
        .file("src/lib.rs", "")
        .file(
            "hello/Cargo.toml",
            r#"
cargo-features = ["edition2024"]

# Before project
[ project ] # After project header
# After project header line
name = "hello"
edition = "2021"
# After project table
"#,
        )
        .file("hello/src/lib.rs", "")
        .build();

    p.cargo("fix --edition --allow-no-vcs")
        .masquerade_as_nightly_cargo(&["edition2024"])
        .with_stderr(
            "\
[MIGRATING] Cargo.toml from 2021 edition to 2024
[FIXED] Cargo.toml (1 fix)
[WARNING] [CWD]/hello/Cargo.toml: `[project]` is deprecated in favor of `[package]`
[LOCKING] 2 packages to latest compatible versions
[CHECKING] foo v0.0.0 ([CWD])
[MIGRATING] src/lib.rs from 2021 edition to 2024
[FINISHED] `dev` profile [unoptimized + debuginfo] target(s) in [..]s
",
        )
        .run();
    assert_eq!(
        p.read_file("Cargo.toml"),
        r#"
cargo-features = ["edition2024"]

[workspace]
members = ["hello"]

# Before project
[ package ] # After project header
# After project header line
name = "foo"
edition = "2021"
# After project table
"#
    );
    assert_eq!(
        p.read_file("hello/Cargo.toml"),
        r#"
cargo-features = ["edition2024"]

# Before project
[ project ] # After project header
# After project header line
name = "hello"
edition = "2021"
# After project table
"#
    );
}
```

</details>